src/Cloud4RPi.cpp

```diff
@@ -0,0 +1,64 @@
+#include "Cloud4RPi.h"
+
+Cloud4RPi::Cloud4RPi(const String &_deviceToken, const String &_server, int _port) :
+    deviceToken(_deviceToken),
+    server(_server),
+    port(_port),
+    mqttClient(NULL) {
+      // Empty
+}
+
+void Cloud4RPi::begin(Client& _client) {
+  mqttClient = new PubSubClient(_client);
+  mqttClient->setServer(server.c_str(), port);
+}
+
+Cloud4RPi::~Cloud4RPi() {
+  if (mqttClient == NULL) {
+    delete mqttClient;
+    mqttClient = NULL;
+  };
+}
+
+bool Cloud4RPi::loop() {
+    return mqttClient->loop();
+}
+
+bool Cloud4RPi::connected() {
+    return mqttClient->connected();
+}
+
+bool Cloud4RPi::ensureConnection(int maxReconnectAttempts, int reconnectTimeout) {
+    int attempt = 0;
+    bool forever = maxReconnectAttempts == RETRY_FOREVER;
```

Comment:
Either this should be changed to `<=` or the doc-comment in the **.h** file should be changed